libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs

```diff
@@ -29,11 +31,71 @@ public static class DialogExtensions
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         public static async Task RunAsync(this Dialog dialog, ITurnContext turnContext, IStatePropertyAccessor<DialogState> accessor, CancellationToken cancellationToken)
         {
-            var dialogSet = new DialogSet(accessor) { TelemetryClient = dialog.TelemetryClient };
+            var dialogSet = new DialogSet(accessor);
             dialogSet.Add(dialog);
 
+            // look for the IBotTelemetryClient on the TurnState, if not there take it from the Dialog, if not there fall back to the "null" default
+            var telemetryClient = turnContext.TurnState.Get<IBotTelemetryClient>() ?? dialog.TelemetryClient ?? new NullBotTelemetryClient();
+            dialog.TelemetryClient = telemetryClient;
+            dialogSet.TelemetryClient = telemetryClient;
+
             var dialogContext = await dialogSet.CreateContextAsync(turnContext, cancellationToken).ConfigureAwait(false);
 
+            // drop the return value - the whole point of this function is that the caller doesn't care
```

Comment:
I would remove this comment, it doesn't really help

Reply:
OK